### PR TITLE
Add documentation of workspace setup for custom messages

### PR DESCRIPTION
**Pull request by mabelzhang**
Add documentation for setting up workspace and compiling custom message types and the bridge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ The bridge is currently implemented in C++ as at the time the Python API for ROS
 Because of this its support is limited to only the message/service types available at compile time of the bridge.
 The bridge provided with the prebuilt ROS 2 binaries includes support for common ROS interfaces (messages/services), such as the interface packages listed in the [ros2/common_interfaces repository](https://github.com/ros2/common_interfaces) and `tf2_msgs`.
 See [the documentation](doc/index.rst) for more details on how ROS 1 and ROS 2 interfaces are associated with each other.
-If you would like to use a bridge with other interfaces (including your own custom types), you will have to build the bridge from source (instructions below).
+If you would like to use a bridge with other interfaces (including your own custom types), you will have to build the bridge from source (instructions below), after building and sourcing your custom types in separate ROS 1 and ROS 2 workspaces. See [the documentation](doc/index.rst) for an example setup.
```
Comment (dirk-thomas) [inline README.md]: Please follow the [style guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#markdown-restructured-text-docblocks): one sentence per line.

File: doc/index.rst [hunk 2]
```diff
@@ -22,10 +22,12 @@ In the third step fields are associated with each other.
 If one of the two associated messages has fields which is are not part of the other message they are being ignored.
 If both messages have fields the other message does not contain it is assumed that the mapping is incomplete and no association is established.
 
+If the package name, message name, and all field names and types are the same between a ROS 1 package and a ROS 2 package, then the messages will automatically be associated with each other without additional specification. The bridge needs to be recompiled, after the ROS 1 and ROS 2 packages have been compiled and sourced, each in a separate workspace, for a total of three workspaces.
```
Comment (dirk-thomas) [inline doc/index.rst]: Please add a note that this is only true for package names with a certain suffix.

File: doc/index.rst [hunk 3]
```diff
@@ -144,3 +146,95 @@ The mapping can optionally only define ``request_fields_1_to_2`` or ``response_f
...
+    . /opt/ros/melodic/setup.bash
+    . <workspace-parent-path>/ros1_msgs_ws/install_isolated/setup.bash
+    # Verify the topic is listed
+    rostopic list
+    rostopic echo /joint_command
+
+Known Issues
+------------
+
+- Currently, ``--bridge-all-topics`` may be needed to bridge correctly. Once the mapping is established with the ROS master, it may be possible to rerun the bridge without bridging all topics, in order to selectively bridge topics. However, this is not guaranteed.
```
Comment (dirk-thomas) [inline doc/index.rst]: > `--bridge-all-topics` may be needed to bridge correctly.

Please describe in which cases that is needed.

**Comment by mabelzhang** [comment]
I messed up the sign-offs here so I'm closing this and will re-open one after addressing the comments.

**Pull request closed without merging**